src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ExtenderProvidedPropertyAttribute.cs

```diff
@@ -52,9 +52,17 @@ public override bool Equals(object obj)
             {
                 return true;
             }
+            if (!(obj is ExtenderProvidedPropertyAttribute other))
+            {
+                return false;
+            }
+
+            if (other.IsDefaultAttribute())
+            {
+                return IsDefaultAttribute();
+            }
```

Comment:
Where did this check come from?

Reply:
Basically the attribute is default if, any only if (because its sealed) one of either `ExtendedProperty`, `Provider` or `ReceiverType` is null. Since the type is sealed, this means that if the other attribute is default then it can never equal a non-default attribute hence avoiding this null reference exception

Reply:
That's really subtle, and would be prone to breaking if, for example, the type were ever unsealed.  What is the null reference exception to which you refer / where does it come from?  There's no more explicit / less-error prone way to check for it?

Reply:
OK - updated to be more clear

Reply:
Thanks